test/configCases/parsing/harmony-this/abc.js

```diff
@@ -22,6 +22,13 @@ export class C {
 	foo() {
 		return this.x;
 	}
+	bar(x = this.x) {
+		return x;
+	}
+}
+
+export const extendC = () => {
+	return class extends this.C {};
```

Comment:
I guess this would fail even without the change, since this.C is not a class. Maybe do this: `extends from this.Buffer`.